File: src/main/java/plugin/GDScriptFileTypeFactory.java
package plugin;

import com.intellij.openapi.fileTypes.FileTypeConsumer;
import com.intellij.openapi.fileTypes.FileTypeFactory;
import org.jetbrains.annotations.NotNull;

public class GDScriptFileTypeFactory extends FileTypeFactory {
	@Override
	public void createFileTypes(@NotNull FileTypeConsumer fileTypeConsumer) {
		fileTypeConsumer.consume(GDScriptFileType.INSTANCE, GDScriptFileType.FILE_EXTENSION);
	}
}

File: src/main/kotlin/gdscript/GDScriptFileTypeFactory.kt
package gdscript

import com.intellij.openapi.fileTypes.FileTypeConsumer
import com.intellij.openapi.fileTypes.FileTypeFactory

class GDScriptFileTypeFactory : FileTypeFactory() {

    override fun createFileTypes(fileTypeConsumer: FileTypeConsumer) {
        fileTypeConsumer.consume(GDScriptFileType, GDScriptFileType.defaultExtension)
    }

}
